grammar.js

```diff
@@ -535,8 +547,10 @@ module.exports = grammar({
         seq(receiver, arguments),
         seq(receiver, prec(PREC.CURLY_BLOCK, seq(arguments, block))),
         seq(receiver, prec(PREC.DO_BLOCK, seq(arguments, doBlock))),
+        prec(PREC.CURLY_BLOCK, seq(receiver, $.array, block)),
+        prec(PREC.DO_BLOCK, seq(receiver, $.array, doBlock)),
```

Comment:
This will cause `call` nodes to have a new type of child (`array`), instead of just receiver, `arguments` and `block`/`do_block`. It seems a little bit counterintuitive to me, so I'd like to try and avoid this.